### ISSUE TITLE
False positive in "Static lambdas binding" with injected languages

### ISSUE THREAD

**Issue by FlyingDR** [kalessil/phpinspectionsea#1382]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), v3.0.2 |
| Language level | 7.1  |

Current behaviour
--
"Static lambdas binding" inspection seems to ignore a fact that closure code may include fragments of other languages. Consider this simplified example of Wordpress shortcode handler that tries to add inline JavaScript:
```php
add_shortcode('my-shortcode', static function () {
    wp_scripts()->add_inline_script('my-js', <<<JS
        $('.my-class').each(function () {
            var $this = $(this);    // Error have been reported for $this
            // Remaining script
        });
JS
    );
});
```
Plugin incorrectly treats `$this` as PHP code while it is actually JavaScript (and PHPStorm itself recognizes it as JavaScript). Here is actual screenshot:
![image](https://user-images.githubusercontent.com/822673/62776266-b9766580-bab2-11e9-9533-ba8888978995.png)
 
Expected behaviour
--
There should be no error report in this case.

I don't know internals of JetBrains products, but hopefully that can inform plugin about injected fragments of other languages and plugin will be able to ignore pieces of non-PHP code while running inspections.

Environment details
---
PHPStorm 2019.2 on Windows 7, Php Inspections (EA Ultimate), v3.0.2

**Comment by kalessil** [comment]
Well, the plugin attaches to PHP code only, in this case the IDE considering the code fragment to be PHP. I think the easiest way to solve this to follow "Help -> Contact support", perhaps PhpStorm folk can advice something for this case.

**Comment by FlyingDR** [comment]
In a case if somebody will get into similar situation: both PHPStorm and Php Inspections plugin behaves correctly in this scenario. 

Reason of the issue was caused by incorrect use of [Heredoc](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc) which allows embedding of PHP variables while [Nowdoc](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.nowdoc) was necessary in this case. 

In other words it was necessary to use `<<<'JS'` instead of `<<<JS`
